core/src/tangram.h

```diff
@@ -184,29 +181,8 @@ class Map {
 
 private:
 
-    enum class EaseField { position, zoom, rotation, tilt };
-
-    void setEase(EaseField _f, Ease _e);
-    void clearEase(EaseField _f);
-    void setScene(std::shared_ptr<Scene>& _scene);
-    void setPositionNow(double _lon, double _lat);
-    void setZoomNow(float _z);
-    void setRotationNow(float _radians);
-    void setTiltNow(float _radians);
-
-    std::mutex m_tilesMutex;
-    std::mutex m_sceneMutex;
-    std::unique_ptr<TileWorker> m_tileWorker;
-    std::unique_ptr<TileManager> m_tileManager;
-    std::shared_ptr<Scene> m_scene;
-    std::shared_ptr<Scene> m_nextScene;
-    std::shared_ptr<View> m_view;
-    std::unique_ptr<Labels> m_labels;
-    std::unique_ptr<InputHandler> m_inputHandler;
-    std::unique_ptr<AsyncWorker> m_asyncWorker;
-    std::vector<SceneUpdate> m_sceneUpdates;
-    std::array<Ease, 4> m_eases;
-    bool m_cacheGlState;
+    class Impl;
+    std::unique_ptr<Impl> impl;
```

Comment:
Just a note here that `Map::Impl` class has its move constructor suppressed, unless explicitly defined/defaulted. This might have an affect if the user of the `Map` class tries to `move` an instance.
Reference: http://en.cppreference.com/w/cpp/language/pimpl